File: Day15_Pascal2.py
def getRow(self, rowIndex: int) -> list[int]:
    ans = [1]
    if(rowIndex >= 1):
        ans = [1, 1]
    if(rowIndex >= 2):
        for j in range(2, rowIndex + 1):
            l = []
            l.append(1)
            for i in range(1, len(ans)):
                l.append(ans[i] + ans[i-1])
            l.append(1)
            ans = l
    return ans

File: Day15_Pascal2.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> getRow(int rowIndex)
{
    vector<int> ans = {1};
    if (rowIndex >= 1)
    {
        ans = {1, 1};
    }
    if (rowIndex >= 2)
    {
        for (int j = 2; j <= rowIndex; j++)
        {
            vector<int> v;
            v.push_back(1);
            for (int i = 1; i < ans.size(); i++)
            {
                v.push_back(ans[i] + ans[i - 1]);
            }
            v.push_back(1);
            ans = v;
        }
    }
    return ans;
}

int main()
{
    auto v = getRow(3);
    for (auto &i : v)
    {
        cout << i << " ";
    }
    cout << endl;
    return 0;
}